ponys.js

```diff
@@ -18,6 +19,7 @@ export default class {
 		url = new URL(url, location.href);
 
 		let script = template.querySelector('script[setup]') || template.querySelector('script');
+		if (script && (!script.hasAttribute("setup") || script.type != "module")) console.warn("setup & type=module attributes expected");
```

Comment:
I think I'd be rather inclined to go straight for the removal of the non-`setup` option, and merge this in for v0.4.0:
```suggestion
		let script = template.querySelector('script[setup]');
```
As for the warning regarding `type="module"`: given that `setup` is an entirely new attribute invented for this purpose (inspired by Vue, of course), there's no reason that it couldn't also imply that the value of the `type` attribute is `"module"` - much like how `type="module"` implies `defer`! - so I don't feel like being too strict with users about that.

Reply:
Regarding the `type=module` part, this is more so a suggestion from how this feature would eventually be supported natively. Simply put, I'm not aware of any native way in browsers today where you can have a `<script>` tag act as an ES6 Module without specifying `type=module` (correct me if I'm wrong). Sure, future spec could say if `setup` attribute is present then assume `type=module`, but IMO that's more-or-less a opinion made by this library and unlikely to be made at a spec level.

My thought process was to make sure to encourage the user on the right path for future native/spec-driven support. 

I've never worked with Vue directly, but I did see in their [docs](https://vuejs.org/guide/extras/web-components#tips-for-a-vue-custom-elements-library) where they only use the setup tag, as you preluded too. And in that context, I can see why you went this way. From my perspective, I thought it was very strange to not have to put the `type=module`, since I was thinking about this from the perspective of native JS/HTML and no other front end frameworks.

Just my two-cents, I have no gripe with it remaining as it is. I also support the enforcement of requiring the user to use the `setup` tag at minimum.